app/importers/mahonia_mapper.rb

```diff
@@ -28,9 +28,27 @@ def fields
     end
   end
 
+  def creator_name(first_name: first, last_name: last)
+    "#{first_name} #{last_name}".strip
+  end
+
+  # return first and last names of 1-n authors
+
   def creator
-    return if metadata['author1_fname'].nil? && metadata['author1_lname'].nil?
-    Array("#{metadata['author1_fname']} #{metadata['author1_lname']}")
+    creators = []
+
+    number_of_authors = metadata.keys.select do |k|
+      k.match?(/^author[0-9]*_lname$/)
+    end.count
+
+    unless metadata["author1_fname"].nil? && metadata["author1_lname"].nil?
```

Comment:
If `author1_fname` and `author1_lname` are both `nil`, but there's a valid `author2_*`, `author3_*`, etc... do we skip the remaining authors?

Reply:
@no-reply we could definitely check for the others, but if their exported numbering scheme skipped over 1, that sounds like an error. But I guess it would be best to just take whatever data is available.

Reply:
Yeah. I think a more complete handling will serve us better in the long run, but I'm okay with cutting the corner here if we're pretty sure it's not going to be an issue. 

I'm okay with this as is, if it seems good enough to you. Just wanted to make sure you had considered your options.